WooCommerce/Classes/Authentication/Jetpack Setup/Native Jetpack Setup/LoginJetpackSetupView.swift

```diff
@@ -0,0 +1,174 @@
+import SwiftUI
+
+/// Hosting controller for `LoginJetpackSetupView`.
+///
+final class LoginJetpackSetupHostingController: UIHostingController<LoginJetpackSetupView> {
+    init(siteURL: String, connectionOnly: Bool) {
+        let viewModel = LoginJetpackSetupViewModel(siteURL: siteURL, connectionOnly: connectionOnly)
+        super.init(rootView: LoginJetpackSetupView(viewModel: viewModel))
+    }
+
+    @available(*, unavailable)
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureNavigationBarAppearance()
+    }
+
+    /// Shows a transparent navigation bar without a bottom border.
+    private func configureNavigationBarAppearance() {
+        let appearance = UINavigationBarAppearance()
+        appearance.configureWithTransparentBackground()
+        appearance.backgroundColor = .systemBackground
+
+        navigationItem.standardAppearance = appearance
+        navigationItem.scrollEdgeAppearance = appearance
+        navigationItem.compactAppearance = appearance
+
+        let title = NSLocalizedString("Cancel", comment: "Button to dismiss the site credential login screen")
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: title, style: .plain, target: self, action: #selector(dismissView))
+    }
+
+    @objc
+    private func dismissView() {
+        dismiss(animated: true)
+    }
+}
+
+/// View to show the process of Jetpack setup during login.
+///
+struct LoginJetpackSetupView: View {
+    @ObservedObject private var viewModel: LoginJetpackSetupViewModel
+
+    /// Scale of the view based on accessibility changes
+    @ScaledMetric private var scale: CGFloat = 1.0
+
+    init(viewModel: LoginJetpackSetupViewModel) {
+        self.viewModel = viewModel
+    }
+
+    private var title: String {
+        viewModel.connectionOnly ? Localization.connectingJetpack : Localization.installingJetpack
+    }
+
+    /// Attributed string for the description text
+    private var descriptionAttributedString: NSAttributedString {
+        let font: UIFont = .body
+        let boldFont: UIFont = font.bold
+        let siteName = viewModel.siteURL.trimHTTPScheme()
+
+        let attributedString = NSMutableAttributedString(
+            string: String(format: Localization.description, siteName),
+            attributes: [.font: font,
+                         .foregroundColor: UIColor.text.withAlphaComponent(0.8)
+                        ]
+        )
+        let boldSiteAddress = NSAttributedString(string: siteName, attributes: [.font: boldFont, .foregroundColor: UIColor.text])
+        attributedString.replaceFirstOccurrence(of: siteName, with: boldSiteAddress)
+        return attributedString
+    }
```

Comment:
Super nit: Moving these to the view model will help us write unit tests.

Reply:
True! I moved these to the view model in 6c1fc50c22f7dc47216e33a6da5f877582110e5f and make them constant/lazy to avoid unnecessary recalculation.